doc/asic-fw-update/asic-fw-update-config.md

```diff
@@ -0,0 +1,285 @@
+  auto-update     Control ASIC firmware auto update [ enable | disable ]
+  default         Set default firmware source [ image | user ]
+```
+
+#### Config DB Enhancements
+To be able to configure ASIC firmware update flow we need a new ConfigDB table.
+Basically, the table is bery simple it will have such structure:
+* CONFIG_DB
+  * FIRMWARE
+    * asic
+      * auto_update
+      * default 
+
+Here is a YANG model of a new table:
+```
+module sonic-firmware {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-firmware";
+    prefix firmware;
+
+    description "FIRMWARE YANG Module for SONiC-based OS";
+
+    revision 2022-05-05 {
+        description "First Revision";
+    }
+
+    typedef auto-update {
+        description "Configuration to set if firmware is going to be updated
+                     automatically";
+        type string {
+            pattern "enable|disable";
+        }
+    }
+
+    typedef default-source {
+        description "Configuration to set firmware source: system image or 
+                     user-defined";
+        type string {
+            pattern "image|user";
+        }
+    }
+
+    container sonic-firmware {
+
+        container FIRMWARE {
+
+            description "FIRMWARE part of config_db.json";
+
+            container asic {
+
+                leaf auto_update {
+                    type auto-update;
+                    default "enable";
+                }
+
+                leaf default {
+                    type default-source;
+                    default "image";
+                }
+
+            }
+
+        }
+        /* end of container FIRMWARE */
+    }
+    /* end of top level container */
+}
+/* end of module nvue */
+```
+
+#### Feature configuration
+<!-- omit in toc -->
+##### Disable FW auto-update
+```
+redis-cli -n 4 HSET "FIRMWARE|asic" "auto_update" "disable"
+sudo reboot
+```
+<!-- omit in toc -->
+##### Enable auto-update, change source to user-defined
+```
+redis-cli -n 4 HSET "FIRMWARE|asic" "auto_update" "enable"
+redis-cli -n 4 HSET "FIRMWARE|asic" "default" "user"
+sudo mkdir -p /host/mlnx/asic/
+sudo cp /tmp/fw-SPC2-rel-29_2010_2270-EVB.mfa /host/mlnx/asic/fw.mfa
+sudo reboot
+```
+
+### Multi-DB support
+Application should be using swss-common library or swsssdk which take care of discovering database instances.
+
+### Configuration Reload
+*config reload* & *config load_minigraph* are used to clear current
+configuration and import new configuration from the input file
+or from /etc/sonic/config_db.json. This command shall stop all services before
+clearing the configuration and then restarts those services.
+Thus, any service that consumes CONFIG DB data has to be restarted on reload
+commands.
+This new feature does not subsribe to the changes in ConfigDB, it reach the
+configuration on demand, so it won't be affected by configuration reload.
+
+### Multi-ASIC
+As you may notice there is no ASIC id for new database table here. The reason
```

Comment:
This is a very broad assumption that multiasics will all run the same firmware. If I have a multi asic platform with different flavors of ASIC there might be a possibility of them requiring different firmware. I recommend you to mention that currently the support is for single-ASIC and multi-ASIC is out of scope.